app/src/main/java/org/feature/fox/coffee_counter/ui/CoreActivity.kt

```diff
@@ -41,5 +52,78 @@ class CoreActivity : ComponentActivity() {
                 }
             }
         }
+        mNfcAdapter = NfcAdapter.getDefaultAdapter(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (mNfcAdapter != null) {
+            val options = Bundle()
+            // Work around for some broken Nfc firmware implementations that poll the card too fast
+            options.putInt(NfcAdapter.EXTRA_READER_PRESENCE_CHECK_DELAY, 250)
+
+            // Enable ReaderMode for all types of card and disable platform sounds
+            mNfcAdapter!!.enableReaderMode(
+                this,
+                this,
+                NfcAdapter.FLAG_READER_NFC_A or
+                        NfcAdapter.FLAG_READER_NFC_B or
+                        NfcAdapter.FLAG_READER_NFC_F or
+                        NfcAdapter.FLAG_READER_NFC_V or
+                        NfcAdapter.FLAG_READER_NFC_BARCODE or
+                        NfcAdapter.FLAG_READER_NO_PLATFORM_SOUNDS,
+                options
+            )
+        }
+    }
+
+    override fun onPause() {
+        super.onPause()
+        if (mNfcAdapter != null) mNfcAdapter!!.disableReaderMode(this)
+    }
+
+    override fun onTagDiscovered(tag: Tag) {
+        val mNdef: Ndef? = Ndef.get(tag)
+
+        if (mNdef == null){
+            runOnUiThread {
+                Toast.makeText(
+                    applicationContext,
+                    "something went wrong",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }else{
```

Comment:
Format your files with autoformat please.

Reply:
If you are talking about ctrl+alt+L, it's formated that way bc I used that

Reply:
Hmm, maybe you've changed your settings for formatting?
`}else{` instead of `} else {` seems to be rather rare.